build.gradle.kts

```diff
@@ -0,0 +1,132 @@
+import org.jetbrains.intellij.tasks.PatchPluginXmlTask
+import org.jetbrains.intellij.tasks.PublishPluginTask
+import org.jetbrains.intellij.tasks.RunIdeTask
+import org.jetbrains.intellij.tasks.RunPluginVerifierTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.util.EnumSet
+
+plugins {
+  id("idea")
+  id("java")
+  kotlin("kapt") version "1.8.10"
+  id("org.jetbrains.intellij") version "1.13.3"
+  id("org.jetbrains.kotlin.jvm") version "1.8.10"
+  id("org.jetbrains.kotlin.plugin.serialization") version "1.4.32"
+}
+
+java {
+  sourceCompatibility = JavaVersion.VERSION_17
+  targetCompatibility = JavaVersion.VERSION_11
+}
+
+// The latest supported versions. Note, these are updated automatically from update-major-versions.sh
+val IIC_RELEASE_VERSION = "231.9011.34"
+val IIC_EAP_VERSION = "232.6095.10"
+val GO_RELEASE_VERSION = "231.9011.34"
+val GO_EAP_VERSION = "232.6095.10"
+// The oldest supported versions.
+val IIC_FROM_VERSION = "222.4554.10"
+val GO_FROM_VERSION = "222.4554.12"
+
+data class Product(
+  val name: String, // identifier for this product
+  val sdkVersion: String, // the version string passed to the intellij sdk gradle plugin
+  val goPluginVersion: String, // a specific version for the go plugin
+)
+
+val products = listOf(
+  Product(name = "release", sdkVersion = IIC_RELEASE_VERSION, goPluginVersion = IIC_RELEASE_VERSION),
+  Product(name = "eap", sdkVersion = "LATEST-EAP-SNAPSHOT", goPluginVersion = IIC_EAP_VERSION),
+)
+val product = products.first { it.name == (System.getenv("PRODUCT_NAME") ?: "eap") }
```

Comment:
1. Should this throw an exception if the `.getenv` fails?
2. Could we have seperate build tasks created by enumerating `products`?

Reply:
> Should this throw an exception if the `.getenv` fails?
No; with the IJ plugin there should be a default one to use when developing locally. I can check if there's a neater way to select a version.

> Could we have seperate build tasks created by enumerating `products`?
It's quite a bit more work to set up (and the multiple products is mostly only interesting when publishing). The task configuration is done via the IJ plugin, and IIUC that plugin assumes you're working with one version of IJ.